### PR TITLE
[sphinx] Adds Logging section in Developer Guide

### PR DESCRIPTION
**Pull request by matteobarbera**
I've written a page to detail some of the logging options that are available for Paparazzi. Not everything that I saw on the Paparazzi Wiki made it here because from my understanding most of the other options will soon become deprecated. 

I wasn't 100% sure about the information on the Pprzlog format so comments on that section are welcome.

### CODE REVIEW

File: doc/sphinx/source/developer_guide/logging.rst [hunk 1]
```diff
@@ -0,0 +1,264 @@
...
+.. code-block:: C
+
+  #define PRINT(string,...) fprintf(stderr, "[orange_avoider->%s()] " string,__FUNCTION__ , ##__VA_ARGS__)
+
+This allows you to only have to write ``PRINT("<your debug message>")`` in your code and it will automatically
+prepend ``"[orange_avoider->%s()] "`` to your message, which can help debugging if multiple modules
+are printing information simultaneously.
+
+When running in simulation ``fprintf`` automatically prints the message in the Paparazzi Center. You can also
+read these messages on the real drone using the following steps:
```
Comment (gautierhattenberger) [inline doc/sphinx/source/developer_guide/logging.rst:43]: I suggest to specify that this example is only valid for Bebop 1 et 2.

File: doc/sphinx/source/developer_guide/logging.rst [hunk 2]
```diff
@@ -0,0 +1,264 @@
...
+Logging on ChibiOS systems can be achieved using the ``flight_recorder`` module, which depends on a
+few other modules that also need to be included in your airframe.
+
+.. code-block:: xml
+
+  <firmware name="rotorcraft">
+    <module name="tlsf"/>
+    <module name="pprzlog"/>
+    <module name="logger" type="sd_chibios"/>
+    <module name="flight_recorder"/>
```
Comment (gautierhattenberger) [inline doc/sphinx/source/developer_guide/logging.rst:106]: With the new dependency solver, it is only necessary to load `flight_recorder` module.

Also, when using `sd_chibios` only, a log file is opened and can be used to log any type of data in ASCII (or even binary if needed) like a regular file. When loading the flight recorder module, two separated files are opened. Have a look at the `meteo_stick` module for an example of using the different options depending of the module's configuration.

File: doc/sphinx/source/developer_guide/logging.rst [hunk 3]
```diff
@@ -0,0 +1,264 @@
...
+.. tip::
+
+  You can add the ``export`` lines above to your ``.bashrc`` file to automatically set the environment variables
+  every time a new Terminal is opened.
+
+Once the environment variables are set you can run ``sd2log`` in your terminal:
+
+.. code-block:: bash
+
+  ~/paparazzi/sw/logalizer/sd2log fr_XXXX.LOG
```
Comment (gautierhattenberger) [inline doc/sphinx/source/developer_guide/logging.rst:171]: `~/paparazzi/sw/logalizer/sd2log fr_XXXX.LOG output_dir_path` to extract the log in an other folder than default log path

File: doc/sphinx/source/developer_guide/logging.rst [hunk 4]
```diff
@@ -0,0 +1,264 @@
...
+
+Pprzlog format
+-----------------
+
+The Pprzlog format creates multiple files that can be used to analyse flight data and replay the flight. A log is
+split into the following files:
+
+- A ``.log`` file, an XML file, which contains a copy of the whole configuration (airframes, flight plans, ...)
+- A ``.data`` file, an ascii file, which contains the list of the received messages. Each message is time-stamped in
+  milliseconds since the creation of the file and marked with the ID of the sending aircraft
```
Comment (gautierhattenberger) [inline doc/sphinx/source/developer_guide/logging.rst:192]: actually a floating point value in seconds with 3 digits

File: doc/sphinx/source/developer_guide/logging.rst [hunk 5]
```diff
@@ -0,0 +1,264 @@
...
+- A ``.tlm`` file, ???
```
Comment (gautierhattenberger) [inline doc/sphinx/source/developer_guide/logging.rst:193]: the `tlm` file is a copy of the original `.LOG` file, with the same naming `data_time.*`. It makes it easier to associate with decoded log files.
By the way, a recommended use is to directly call `sd2log` on the file still on the SD card mounted as a mass storage, it is not really needed to copy it first on the computer.

File: doc/sphinx/source/developer_guide/logging.rst [hunk 6]
```diff
@@ -0,0 +1,264 @@
...
+
+  <message name="ATTITUDE" id="6">
+    <field name="phi"   type="float" unit="rad" alt_unit="deg"/>
+    <field name="psi"   type="float" unit="rad" alt_unit="deg"/>
+    <field name="theta" type="float" unit="rad" alt_unit="deg"/>
+  </message>
+
+In this case, at the time the message was logged, the attitude of the drone corresponded to :math:`{\phi} = 0.036228`,
+:math:`{\theta} = 0.018550`, and :math:`{\psi} = 0.021443` radians. Note that the appropriate ``messages.xml``
+description, i.e. the one which has been used while th elog was created, is itself stored in the associated ``.log``
```
Comment (gautierhattenberger) [inline doc/sphinx/source/developer_guide/logging.rst:216]: `the log`

**Review by gautierhattenberger — CHANGES_REQUESTED** [review]

File: doc/sphinx/source/developer_guide/logging.rst (hunk 1 above)
Comment (matteobarbera) [reply doc/sphinx/source/developer_guide/logging.rst:43]: Is this not possible with every Linux based OS that has WiFi and specifies a host for a telnet connection? E.g. the Disco, although during flight you wouldn't be able to maintain WiFi connection

**Comment by matteobarbera** [comment]
I've addressed the changes (except for the Bebop 1/2 comment, I thought it worked with every Linux based OS that has WiFi and that specifies a host) and added a section on SD ChibiOS logger covering how to create custom log functions for binary and ASCII logs. 

I used `meteo_stick.c` as a reference as suggested, but as I've never attempted to do something like this it's very possible that something is incorrect in my explanation  :sweat_smile:

File: doc/sphinx/source/developer_guide/logging.rst (hunk 1 above)
Comment (gautierhattenberger) [reply doc/sphinx/source/developer_guide/logging.rst:43]: It's true that it works for Disco which is sharing the same hardware than Bebop more or less. The only Linux-based system that we support are Parrot drones, so I guess it is fine even if there are so things specific to these platforms, like default IP or directory tree.

**Review by gautierhattenberger — APPROVED** [review]

**Pull request merged**